### PR TITLE
chatqna kubernetes readme.

### PR DESCRIPTION
**Pull request by mkbhanda**
Signed-off-by: mkbhanda <malini.bhandaru@intel.com>

## Description

The summary of the proposed changes as long as the relevant motivation and context.

## Issues

List the issue or RFC link this PR is working on. If there is no such link, please mark it as `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

**Review by daisy-ycguo — COMMENTED** [review]
Thank you for handling this. 
I wonder if you would like to change the README under this folder `ChatQnA/kubernetes/` as an overall introduction of `Deploy ChatQnA in Kubernetes Cluster on Xeon and Gaudi`, other than `ChatQnA/kubernetes/manifest`?

**Comment by mkbhanda** [comment]
> Thank you for handling this. I wonder if you would like to change the README under this folder `ChatQnA/kubernetes/` as an overall introduction of `Deploy ChatQnA in Kubernetes Cluster on Xeon and Gaudi`, other than `ChatQnA/kubernetes/manifest`? The GMC pipeline yaml files also go to this folder `ChatQnA/kubernetes/` according to PR #334

I thought about this quite a bit and decided to just keep one README in manifests because the top level chatqna readme referred to one in the manifests directory. The Kubernetes directory will need revisiting in the context of some changes coming and we can revisit this item them.

File: ChatQnA/kubernetes/manifests/README.md [hunk 1]
```diff
@@ -14,63 +23,80 @@ You should have all the images
...
-
-- Be sure to modify HUGGINGFACEHUB_API_TOKEN and other important values in qna_configmap_guadi.yaml and qna_configmap_xeon.yaml
-- Be sure the node has path /mnt/models to store all the models
+```sh
+kubectl create ns chatqa
+kubectl apply -f $(pwd)/chatQnA_xeon.yaml
+```
 
-### Deploy
+**GMC will reconcile chatQnA custom resource and get all related components/services ready**
```
Comment (daisy-ycguo) [inline ChatQnA/kubernetes/manifests/README.md:43]: I think there is a step missed here to create a configmap object, refer to [GMC readme](https://github.com/opea-project/GenAIInfra/tree/main/microservices-connector#to-deploy-on-the-cluster) in GenAIInfra.
```
Create ConfigMap for GMC to hold GenAI Components and GMC Router manifests
kubectl create configmap gmcyaml -n $SYSTEM_NAMESPACE --from-file $(pwd)/config/manifests
```

Comment (mkbhanda) [reply ChatQnA/kubernetes/manifests/README.md:43]: Thank you for the careful review @daisy-ycguo  :-). The config map gets created when the user installs GMC by following the link to instructions. This config map is common to all pipelines and adequate to create once.
Xie xie.

File: ChatQnA/kubernetes/manifests/README.md [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
 
-You should have all the images
+Please install GMC  in your Kubernetes cluster, if you have not already done so, by following the steps in Section "Getting Started" at [GMC Install](https://github.com/opea-project/GenAIInfra/tree/main/microservices-connector). Soon as we publish images to Docker Hub, at which point no builds will be required, simplifying install/
+
+
+The ChatQnA application is defined as a Custom Resource (CR) file that the above GMC operator acts  upon. It first checks if the microservices listed in the CR yaml file are running, if not starts them and then proceeds to connect them. When the ChatQnA RAG pipeline is ready, the service endpoint details are returned, letting you use the application. Should you use "kubectl get pods" commands you will see all the component microservices, in particular `embedding`, `retriever`, `rerank`, and `llm`.
+
+
+## Uses prebuilt images
```
Comment (ashahba) [inline ChatQnA/kubernetes/manifests/README.md:12]: `Uses prebuilt images` -> `Using prebuilt images`

File: ChatQnA/kubernetes/manifests/README.md [hunk 3]
```diff
@@ -14,63 +23,80 @@ You should have all the images
...
 
+Should you desire to use the Gaudi accelerator, two alternate images are used for the embedding and llm services.
 For Gaudi:
 
 - tei-embedding-service: opea/tei-gaudi:latest
 - tgi-service: ghcr.io/huggingface/tgi-gaudi:1.2.1
 
 > [NOTE]  
-> Please refer README [for Xeon](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/docker/xeon/README.md) and [for Gaudi](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/docker/gaudi/README.md) to build the opea images
+> Please refer README [for Xeon](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/docker/xeon/README.md) and [for Gaudi](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/docker/gaudi/README.md) to build the opea images. These too will be available on Docker Hub soon.
```
Comment (ashahba) [inline ChatQnA/kubernetes/manifests/README.md:33]: `Please refer` -> `Please refer to`

File: ChatQnA/kubernetes/manifests/README.md [hunk 4]
```diff
@@ -14,63 +23,80 @@ You should have all the images
...
-- Be sure the node has path /mnt/models to store all the models
+```sh
+kubectl create ns chatqa
+kubectl apply -f $(pwd)/chatQnA_xeon.yaml
+```
 
-### Deploy
+**GMC will reconcile chatQnA custom resource and get all related components/services ready**
 
+```sh
```
Comment (ashahba) [inline ChatQnA/kubernetes/manifests/README.md:45]: I'd change this to `bash` for consistency

File: ChatQnA/kubernetes/manifests/README.md [hunk 5]
```diff
@@ -14,63 +23,80 @@ You should have all the images
...
+kubectl get service -n chatqa
 ```
-# Deploy Services with Xeon
 
-$ cd ${RepoPath}/ChatQnA/kubernetes/manifests/
-$ ./install_all_xeon.sh
+**Check GMC chatQnA custom resource to get access URL for the pipeline**
+
+```bash
+$kubectl get gmconnectors.gmc.opea.io -n chatqa
```
Comment (ashahba) [inline ChatQnA/kubernetes/manifests/README.md:52]: Remove `$`

File: ChatQnA/kubernetes/manifests/README.md [hunk 6]
```diff
@@ -14,63 +23,80 @@ You should have all the images
...
+NAME     URL                                                      READY     AGE
+chatqa   http://router-service.chatqa.svc.cluster.local:8080      8/0/8     3m
 ```
 
-### Undeploy
+**Deploy one client pod for testing the chatQnA application**
 
+```bash
+kubectl create deployment client-test -n chatqa --image=python:3.8.13 -- sleep infinity
```
Comment (ashahba) [inline ChatQnA/kubernetes/manifests/README.md:60]: Why `python:3.8.13` and not a new one from `python:3.10` ?

File: ChatQnA/kubernetes/manifests/README.md [hunk 7]
```diff
@@ -14,63 +23,80 @@ You should have all the images
...
 ```
-# Remove Services with Xeon
 
-$ ./remove_all_xeon.sh
+**Access the pipeline using the above URL from the client pod**
+
+```bash
+export CLIENT_POD=$(kubectl get pod  -l app=client-test -o jsonpath={.items..metadata.name})
```
Comment (ashahba) [inline ChatQnA/kubernetes/manifests/README.md:66]: one empty space between `pod` and `-l` is encouraged.

File: ChatQnA/kubernetes/manifests/README.md [hunk 8]
```diff
@@ -14,63 +23,80 @@ You should have all the images
...
+kubectl apply -f $(pwd)/chatQnA_xeon.yaml
 ```
 
-### Undeploy
+**Check the tgi-svc-deployment has been changed to use the new LLM Model**
 
-```
-# Remove Services with Xeon
-$ ./remove_all_gaudi.sh
+```sh
```
Comment (ashahba) [inline ChatQnA/kubernetes/manifests/README.md:90]: `sh` -> `bash`

File: ChatQnA/kubernetes/manifests/README.md [hunk 9]
```diff
@@ -14,63 +23,80 @@ You should have all the images
...
+kubectl get deployment tgi-svc-deployment -n chatqa -o jsonpath="{.spec.template.spec.containers[*].env[?(@.name=='LLM_MODEL_ID')].value}"
 ```
 
-## Verify Services
+**Access the updated pipeline using the above URL from the client pod**
 
+```bash
+kubectl exec "$CLIENT_POD" -n chatqa -- curl $accessUrl  -X POST  -d '{"text":"What is the revenue of Nike in 2023?","parameters":{"max_new_tokens":17, "do_sample": true}}' -H 'Content-Type: application/json'
```
Comment (ashahba) [inline ChatQnA/kubernetes/manifests/README.md:97]: one empty space before `-X` and `-d` is recommended.

**Review by ashahba — CHANGES_REQUESTED** [review]
Thanks @mkbhanda for this PR!
Does it make more sense to land this `README.md` under `ChatQnA/kubernetes/README.md` rather than `ChatQnA/kubernetes/manifests/README.md`?

If not I think we should delete this empty README file for the time being:
`ChatQnA/kubernetes/README.md`

**Review by ashahba — APPROVED** [review]
It'll be nice to delete this file: `ChatQnA/kubernetes/README.md`
If not this one is still LGTM.

**Comment by mkbhanda** [comment]
> It'll be nice to delete this file: `ChatQnA/kubernetes/README.md` If not this one is still LGTM.

will do in another commit, a few more readme files to come.

File: ChatQnA/kubernetes/manifests/README.md (hunk 9 above)
Comment (mkbhanda) [reply ChatQnA/kubernetes/manifests/README.md]: fixed

File: ChatQnA/kubernetes/manifests/README.md (hunk 8 above)
Comment (mkbhanda) [reply ChatQnA/kubernetes/manifests/README.md:93]: all shell commands made consistent

**Comment by hshen14** [comment]
LGTM. Per the discussion, let's add back K8s native right after the milestone release.

**Review by hshen14 — APPROVED** [review]

**Pull request merged**